### PR TITLE
docs: cid conversion

### PR DESCRIPTION
**Pull request by lidel**
This PR adds basic docs about CID conversion (v0→v1, text→binary, and cid→hex).

Inspired by questions like https://github.com/multiformats/js-cid/issues/142

### CODE REVIEW

File: docs/concepts/content-addressing.md [hunk 1]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
 
 You can use the first few bytes of the CID to interpret the remainder of the content address and know how to decode the content after being fetched from IPFS. For more details, check out the [CID specification](https://github.com/ipld/cid). It includes a [decoding algorithm](https://github.com/ipld/cid/blob/ef1b2002394b15b1e6c26c30545fd485f2c4c138/README.md#decoding-algorithm) and links to existing software implementations for decoding CIDs.
 
+If you can't decide between CIDv0 and CIDv1, consider choosing CIDv1 for your new project and opt-in by passing version flag (`ipfs add --cid-version 1`). It is more future-proof and [safe for use in browser contexts](/how-to/address-ipfs-on-web/#subdomain-gateway).
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:46]:
```suggestion
If you can't decide between CIDv0 and CIDv1, consider choosing CIDv1 for your new project and opt in by passing a version flag (`ipfs add --cid-version 1`). This is more future-proof and [safe for use in browser contexts](/how-to/address-ipfs-on-web/#subdomain-gateway).
```

File: docs/concepts/content-addressing.md [hunk 2]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+
+IPFS project will switch to CIDv1 as the new default in the near future.
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:48]:
```suggestion
The IPFS project will switch to CIDv1 as the new default in the near future.
```

File: docs/concepts/content-addressing.md [hunk 3]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+
 ## CID Inspector
 
 It's easy to explore a CID for yourself. Want to pull apart a specific CID's multibase, multicodec, or multihash info? You can use the [CID Inspector](https://cid.ipfs.io/#QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU) or the [CID Info panel in IPLD Explorer](https://explore.ipld.io/#/explore/QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU) (both links launch using a sample CID) for an interactive breakdown of differently-formatted CIDs.
 
 Check out ProtoSchool's [Anatomy of a CID](https://proto.school/anatomy-of-a-cid) tutorial to see how a single file can be represented in multiple CID versions.
 
+## CID conversion
+
+Converting from v0 to v1 enables CID to be represented in one of multibase encodings.
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:59]:
```suggestion
Converting a CID from v0 to v1 enables it to be represented in multibase encodings.
```

File: docs/concepts/content-addressing.md [hunk 4]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+The default for CIDv1 is case-insensitive `base32`, but use of shorter `base36` is encouraged for IPNS names to ensure same text representation on [subdomains](/how-to/address-ipfs-on-web/#subdomain-gateway).
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:60]:
```suggestion
The default for CIDv1 is the case-insensitive `base32`, but use of the shorter `base36` is encouraged for IPNS names to ensure same text representation on [subdomains](/how-to/address-ipfs-on-web/#subdomain-gateway).
```

File: docs/concepts/content-addressing.md [hunk 5]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+new CID('QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR').toV1().toString()
+// → bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi
+```
+
+### Text to binary
+
+CID can be represented as both text and a stream of bytes. The latter may be a better choice when speed and storage efficiency are important.
+
+To convert CIDv1 from text to the binary form, simply read the first character
+and then decode the remainder using encoding specified in the [multibase table](https://github.com/multiformats/multibase#multibase-table).
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:83]:
```suggestion
and then decode the remainder using the encoding specified in the [multibase table](https://github.com/multiformats/multibase#multibase-table).
```

File: docs/concepts/content-addressing.md [hunk 6]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+
+JS users can leverage the [`cids`](https://www.npmjs.com/package/cids) library to get binary version as `Uint8Array`:
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:85]:
```suggestion
JS users can leverage the [`cids`](https://www.npmjs.com/package/cids) library to get a binary version as `Uint8Array`:
```

File: docs/concepts/content-addressing.md [hunk 7]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+
+
+```js
+const CID = require('cids')
+new CID('bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi').bytes
+// → Uint8Array [ 1, 112,  18,  32, 195, 196, 115,  62, ... ]
+```
+
+::: warning Be mindful about parsing CID correctly and avoid shortcuts.
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:94]:
```suggestion
::: warning Be mindful about parsing CIDs correctly. Avoid shortcuts.
```

File: docs/concepts/content-addressing.md [hunk 8]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+
+```javascript
+> cid.toString('base16')
+'f01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a'
+
+> cid.toString('base16').substring(1)
+'01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a' // "cid as hex"
+```
+
+To convert back to a CIDv1, prepend hex value with `f` ([multibase prefix](https://github.com/multiformats/multibase#multibase-table) for lowercase base16).
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:124]:
```suggestion
To convert back to a CIDv1, prepend the hex value with `f` ([multibase prefix](https://github.com/multiformats/multibase#multibase-table) for lowercase base16).
```

File: docs/concepts/content-addressing.md [hunk 9]
```diff
@@ -41,12 +43,100 @@ These leading identifiers also provide forward-compatibility, supporting differe
...
+Use it as-is (it is a [valid CID](https://ipfs.io/ipfs/f01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a)), or convert to different multibase by passing it as an argument to `toString`:
+
+```javascript
+> new CID('f' +'01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a').toString('base32')
+// → bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi
+```
+
+::: tip
+[Subdomain gateways](/how-to/address-ipfs-on-web/#subdomain-gateway) convert paths with custom bases like base16 to base32 or base36 in effort to fit a CID in DNS label:
```
Comment (jessicaschilling) [inline docs/concepts/content-addressing.md:133]:
```suggestion
[Subdomain gateways](/how-to/address-ipfs-on-web/#subdomain-gateway) convert paths with custom bases like base16 to base32 or base36, in an effort to fit a CID in a DNS label:
```

File: docs/concepts/glossary.md [hunk 10]
```diff
@@ -174,6 +174,18 @@ A Merkle Tree is a specific type of hash tree used in cryptography and computer
 
 The Mutable File System (MFS) is a tool built into IPFS that lets you treat files like a normal name-based filesystem. You may add, edit, and remove MFS files while all link updates and hashes are taken care of for you. [More about MFS](https://docs.ipfs.io/concepts/file-systems/#mutable-file-system-mfs)
 
+### Multibase
+
+Multibase is a protocol for disambiguating the encoding of base-encoded (e.g., base32, base36, base64, base58, etc.) binary appearing in text. In IPFS it is used as a prefix specifying the encoding used for the remainder of the CID. [More about Multibase](https://github.com/multiformats/multibase#readme)
+
+### Multicodec
+
+Multicodec is an identifier, indicating the format of the target content. It helps people and software to know how to interpret that content after the content is fetched. In IPFS it is backed by an agreed-upon codec table. It is designed for use in binary representations, such as keys or identifiers (i.e [CIDv1](#cid)). [More about Multicodec](https://github.com/multiformats/multicodec#readme)
```
Comment (jessicaschilling) [inline docs/concepts/glossary.md:183]:
```suggestion
Multicodec is an identifier indicating the format of the target content. It helps people and software know how to interpret that content after the content is fetched. In IPFS, it is backed by an agreed-upon codec table. It is designed for use in binary representations, such as keys or identifiers (i.e [CIDv1](#cid)). [More about Multicodec](https://github.com/multiformats/multicodec#readme)
```

File: docs/concepts/glossary.md [hunk 11]
```diff
@@ -174,6 +174,18 @@ A Merkle Tree is a specific type of hash tree used in cryptography and computer
...
+
+### Multihash
+
+Multihash is a protocol for differentiating outputs from various well-established hash functions, addressing size + encoding considerations. It is useful to write applications that future-proof their use of hashes, and allow multiple hash functions to coexist. [More about Multihash](https://multiformats.io/multihash/).
```
Comment (jessicaschilling) [inline docs/concepts/glossary.md:187]:
```suggestion
Multihash is a protocol for differentiating outputs from various well-established hash functions, addressing size and encoding considerations. It is useful to write applications that future-proof their use of hashes, and it allows multiple hash functions to coexist. [More about Multihash](https://multiformats.io/multihash/).
```

**Review by jessicaschilling — CHANGES_REQUESTED** [review]
Minor grammar/style nits. Thank you!

**Comment by lidel** [comment]
Applied style changes, thank you @jessicaschilling !

**Review by jessicaschilling — APPROVED** [review]

**Comment by johnnymatthews** [comment]
Sorry for the delay on my end getting to this. Looks good! Merging in staging.

**Review by johnnymatthews — APPROVED** [review]

**Pull request merged**